cli/running/instance.go

```diff
@@ -50,6 +50,12 @@ type Instance struct {
 	waitMutex sync.Mutex
 	// done represent whether the instance was stopped.
 	done bool
+	// stateMutex used to avoid a race condition under started and shouldStop fields.
+	stateMutex sync.Mutex
+	// started indicates whether the Watchdog has started an Instance.
+	started bool
+	// shouldStop indicates whether the Watchdog should stop the Instance.
```

Comment:
nit: according to this comment, there is no need to call instance.Stop to stop the instance? Is setting this flag really enough to stop the instance?
Instance is a separate struct. Why do we mention watchdog here?